documentation/smg/08_setting_up_jenkins_ci.md

```diff
@@ -0,0 +1,126 @@
+# Setting Up Jenkins CI
+
+We use machines administered by [ANVIL](https://anvil.softeng-support.ac.uk/)
+to run our continuous integration (CI) jobs. These machines are running
+Jenkins v2.190.3.
+
+We would like to be able to script and commit as much of our CI configuration
+as possible. This makes things re-creatable and locally testable. However,
+there is necessarily some set up that must be done within the GUI in Jenkins.
+It is therefore important that this document is kept up-to-date with the steps
+taken to reach the configuration we use.
+
+We will use the CI for building pull requests and building master every
+evening. Each build will create a Horace package that is shippable to users.
+
+## Overview
+
+Our CI builds follow the following high-level process:
+
+- Jenkins checks out the correct branch
+
+  For pull requests:
+
+  - Webhooks are created in GitHub that notify Jenkins when a pull request event
+  occurs.
+  - When Jenkins receives a notification from GitHub it will checkout out the
+  relevant pull request branch and merge it with master.
+
+  For nightly builds:
+
+  - Jenkins will clone master at a specific time each evening.
+
+- Jenkins loads and runs the [`Jenkinsfile`](./../../tools/build_config/Jenkinsfile)
+located in `tools/build_config`.
+- The `Jenkinsfile` script runs our build script (one for Windows, one for Unix)
+and updates GitHub about the status of the build.
+
+## More Detailed Set Up
+
+### GitHub
+
+The only job required on GitHub is to set up webhooks to notify Jenkins of pull
+requests. These can be created by GitHub repository admins by opening the
+settings tab in the [main repo](https://github.com/pace-neutrons/Horace).
+
+<img src="./images/08_github_settings.png">
+
+Then selecting the `Webhooks` menu item on the left-hand side.
+
+Your webhook should have payload URL
```

Comment:
Having now tried to setup the Herbert instance, how we are using secret tokens needs to be clarified:
- one token per repository so any PR for Herbert will trigger ALL Herbert PR- pipelines, Horace will trigger all Horace jobs, 
- one token for the whole of pace-neutrons so any PR on Herbert or Horace (or any other configured repo) will trigger all the PR- pipelines and additional filtering will done in configuration or script level
- one token per target/repo giving many webhooks with unique tokens.

I think the one per repository option is the most straightforward; we should change the tokens so they are `<repo>-<guid>` making it easier to separate them.

Reply:
One per repo sounds sensible